### PR TITLE
improve _get_files in SSH context

### PR DESCRIPTION
**Pull request by thangckt**
this PR solves problem #471 and have some improvements

## Summary by CodeRabbit

- **New Features**
	- Improved error handling during file write operations to prevent data loss.
	- Enhanced command execution logic for better clarity and maintainability.

- **Refactor**
	- Renamed variables for consistency and clarity across methods.
	- Updated argument names and documentation in the `resources_subfields` method.

- **Style**
	- Modernized code formatting by adopting f-strings for improved readability.

### CODE REVIEW

**Comment by thangckt** [comment]
hi @njzjz 
I did not see your PR when I try to make some change
BTW, this PR can fix the problem

File: dpdispatcher/contexts/ssh_context.py [hunk 1]
```diff
@@ -793,12 +793,16 @@ def write_file(self, fname, write_str):
...
+            with self.sftp.open(temp_fname, "w") as fp:
+                fp.write(write_str)
+            # Rename the temporary file
+            self.block_checkcall(f"mv {shlex.quote(temp_fname)} {shlex.quote(fname)}")
         # sftp.rename may throw OSError
-        self.block_checkcall(
-            "mv {} {}".format(shlex.quote(fname + "~"), shlex.quote(fname))
-        )
+        except OSError as e:
+            print(f"Error writing to file {fname}")
```
Comment (njzjz) [inline dpdispatcher/contexts/ssh_context.py:804]:
```suggestion
            dlog.exception(f"Error writing to file {fname}")
```

File: dpdispatcher/machines/pbs.py [hunk 2]
```diff
@@ -324,11 +323,11 @@ def resources_subfields(cls) -> List[Argument]:
                 dict,
                 [
                     Argument(
-                        "sge_pe_name",
+                        "pe_name",
```
Comment (njzjz) [inline dpdispatcher/machines/pbs.py:326]: For the backward compatibility, please add `alias = ["sge_pe_name"]`. Btw, it seems out of the scope of the PR title.

File: dpdispatcher/contexts/ssh_context.py [hunk 3]
```diff
@@ -945,36 +949,30 @@ def _get_files(self, files, tar_compress=True):
...
-                    ) from e
-                raise e
+            file_list = " ".join([shlex.quote(file) for file in files])
+            tar_cmd = f"tar {tar_command} {shlex.quote(of)} {file_list}"
         else:
-            file_list_file = os.path.join(
-                self.remote_root, ".tmp.tar." + str(uuid.uuid4())
-            )
+            file_list_file = pathlib.PurePath(
+                os.path.join(self.remote_root, f"tmp_tar_{uuid.uuid4()}")
```
Comment (njzjz) [inline dpdispatcher/contexts/ssh_context.py:956]: Why is the leading dot removed? By default, `ls` will not show files starting with `.`, i.e. they are hidden.

File: dpdispatcher/contexts/ssh_context.py [hunk 4]
```diff
@@ -945,36 +949,30 @@ def _get_files(self, files, tar_compress=True):
...
-                )
-            except RuntimeError as e:
-                if "No such file or directory" in str(e):
-                    raise FileNotFoundError(
-                        "Any of the backward files does not exist in the remote directory."
-                    ) from e
-                raise e
-        # trans
+            # if not os.path.exists(file_list_file):
+            #     raise FileNotFoundError(f"File list was not created at {file_list_file}")
```
Comment (njzjz) [inline dpdispatcher/contexts/ssh_context.py:960]: This can be removed if not used.

**Review by njzjz — CHANGES_REQUESTED** [review]

File: dpdispatcher/machines/pbs.py (hunk 2 above)
Comment (thangckt) [reply dpdispatcher/machines/pbs.py:326]: I forget to include this in previous PR. 
can you show me where to put alias?

Comment (njzjz) [reply dpdispatcher/machines/pbs.py:326]: Please check https://docs.deepmodeling.com/projects/dargs/en/latest/api/dargs.html#dargs.Argument

**Review by njzjz — APPROVED** [review]

**Pull request merged**